### DISCUSSION TITLE
在未来加入更多独占 FDS 控件

### DISCUSSION THREAD

**Discussion by Aris-Offline** [iNKORE-NET/UI.WPF.Modern#104] [Ideas]
> 对周前一次有关 .NET 9 中的 Fluent 样式的讨论的总结。

由于微软将在 .NET 9 加入内置 WPF Fluent 样式，如果微软发挥超常，那么包括 UI.WPF.Modern 和 WPF UI 在内的一众以 FDS 为特点的第三方 UI 库就面临着被取代的风险。

由于我们已经由桌面开发转向 Web 开发为主，因此在 iUWM 的精力将大大缩减，停更这个项目也在我们的考虑之内。

然而我们想尽可能保持 iUWM 的发展和维护，因此我们提出了一个想法：在坚持 FDS 的同时逐步制作独占的控件及（可能的）风格。

这个想法仍不成熟，欢迎各位提出意见！

**Comment by NotYoojun** [comment]
我觉得搬运和移植 CommunityToolkit 内的组件是一个更可行的方案。

**Reply by Aris-Offline** [reply]
> 我觉得搬运和移植 CommunityToolkit 内的组件是一个更可行的方案。

控件可以，但是以 WPF 的渲染性能貌似做不起 CommunityToolkit 那堆 Animation 吧

而且这样貌似不是一个长久之计（虽然足够吃一段时间了）

**Reply by NotYoojun** [reply]
> > 我觉得搬运和移植 CommunityToolkit 内的组件是一个更可行的方案。
> 
> 控件可以，但是以 WPF 的渲染性能貌似做不起 CommunityToolkit 那堆 Animation 吧
> 
> 而且这样貌似不是一个长久之计（虽然足够吃一段时间了）

现在 SettingsCard 和 SettingsExpander 都差不多了，动画的话肯定是尽量缩减。
自己搓控件花的精力比移植大多了。

**Reply by Aris-Offline** [reply]
嗯，移植确实比独作轻松而且有成效。

对于 WPF 这种类型框架的来说 Layout 里面的东西和 Input 里面的 Segmented 都可以优先考虑。

不过像是 TransitionHelper 这种渲染要求大的就只能放弃了

**Comment by NotYoojun** [comment]
这个东西确实应该要搞起来了，像现在的 MessageBox 就是一个非常不错的例子
